cloudferry/lib/os/discovery/nova.py

```diff
@@ -153,13 +155,23 @@ def equals(self, other):
         return True
 
 
-def _list_ephemeral(remote, server):
+def _need_image_membership(srv):
+    image = srv.image
+    if image is None:
+        return False
+    if image.is_public:
+        return False
+    return image.tenant != srv.tenant
+
+
+def _list_ephemeral(remote_executor, server):
     result = []
-    output = remote.sudo('virsh domblklist {instance}',
-                         instance=server.instance_name)
-    if not output.succeeded:
+    try:
+        output = remote_executor.sudo('virsh domblklist {instance}',
+                                      instance=server.instance_name)
+    except remote.RemoteFailure:
         LOG.warning('Unable to get ephemeral disks for server '
-                    '%s, skipping.', server.object_id)
+                    '%s, skipping.', server.object_id, exc_info=True)
```

Comment:
I suggest you to write a debug message with `exc_info` and remove it from warning.